testing/integration/agent_long_test.go

```diff
@@ -0,0 +1,294 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"os"
+	"regexp"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	logrunner "github.com/leehinman/spigot/pkg/runner"
+	"github.com/sajari/regression"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/elastic/elastic-agent-libs/kibana"
+	"github.com/elastic/elastic-agent/pkg/control/v2/cproto"
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/tools"
+	"github.com/elastic/go-sysinfo"
+	"github.com/elastic/go-sysinfo/types"
+	"github.com/elastic/go-ucfg"
+)
+
+type ExtendedRunner struct {
+	suite.Suite
+	info         *define.Info
+	agentFixture *atesting.Fixture
+
+	ESHost string
+}
+
+type ComponentMetrics struct {
+	Memory        MemoryMetrics  `mapstructure:"memstats"`
+	Handles       HandlesMetrics `mapstructure:"handles"`
+	UnixTimestamp int64
+}
+
+// TestComponent is used as a key in our map of component metrics
+type TestComponent struct {
+	Binary   string `mapstructure:"binary"`
+	Dataset  string `mapstructure:"dataset"`
+	ID       string `mapstructure:"id"`
+	CompType string `mapstructure:"type"`
+}
+
+type MemoryMetrics struct {
+	GcNext      uint64 `mapstructure:"gc_next"`
+	MemoryAlloc uint64 `mapstructure:"memory_alloc"`
+	MemorySys   uint64 `mapstructure:"memory_sys"`
+	MemoryTotal uint64 `mapstructure:"memory_total"`
+	RSS         uint64 `mapstructure:"rss"`
+}
+
+type HandlesMetrics struct {
+	Open  int           `mapstructure:"open"`
+	Limit HandlesLimits `mapstructure:"limit"`
+}
+
+type HandlesLimits struct {
+	Hard uint   `mapstructure:"hard"`
+	Soft uint64 `mapstructure:"soft"`
+}
+
+// MetricsSystem is used for windows handles metrics
+type MetricsSystem struct {
+	Handles HandlesMetrics `mapstructure:"handles"`
+}
+
+type processWatcher struct {
+	handle     types.Process
+	pid        int
+	name       string
+	regHandles *regression.Regression
+}
+
+func TestLongRunningAgentForLeaks(t *testing.T) {
```

Comment:
nit: you rename the file to match the test? Something like `long_running_leak_test.go`.